src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,286 @@
+                item.type === "action" ||
+                item.type === "divider"
+              )
+          )
+        ) {
+          console.warn(
+            "NYPL Reservoir Menu: A `type` value is required for each list item."
+          );
+        } else if (listItemsData.some((item) => !item.id || item.id === "")) {
+          console.warn(
+            "NYPL Reservoir Menu: An `id` value is required for each list item."
+          );
+        } else if (
+          listItemsData.some(
+            (item) =>
+              (item.type === "action" || item.type === "group") &&
+              (!item.label || item.label.trim() === "")
+          )
+        ) {
+          console.warn(
+            "NYPL Reservoir Menu: A `label` value is required for all list actions and groups."
+          );
+        } else if (
+          listItemsData.some((item) => item.type === "action" && !item.onClick)
+        ) {
+          console.warn(
+            "NYPL Reservoir Menu: An `onClick` function is required for all actions."
+          );
+        } else if (
+          listItemsData.some(
+            (item) =>
+              item.type === "group" &&
+              (!item.children || item.children.length === 0)
+          )
+        ) {
+          console.warn(
+            "NYPL Reservoir Menu: A `children` array is required for all list groups."
+          );
+        }
+        return true;
+      };
+      validateProps(listItemsData, selectedItem);
+
+      /** Helper function that renders either an Image or an Icon for a menu item. */
+      const renderMedia = (media: Media | null) => {
+        if (media) {
+          if (media.type === "icon") {
+            return <Icon name={media.name} size="medium" />;
+          } else {
+            return (
+              <Image
+                src={media.src}
+                alt={media.alt}
+                size="xxxsmall"
+                sx={{ borderRadius: "24px" }}
+              />
+            );
+          }
+        }
+        return null;
+      };
+
+      /** Renders the trigger button for the Menu List to open or close. */
+      const getButton = (isOpen) => (
+        <MenuButton
+          sx={styles.menuButton}
+          backgroundColor={isOpen ? "ui.link.primary-05 !important" : "unset"}
+        >
+          {showLabel && (
+            <>
+              <span style={{ paddingRight: "8px" }}>{labelText}</span>
+              <Icon
+                name="arrow"
+                iconRotation={isOpen ? "rotate180" : "rotate0"}
+                size="xsmall"
+              />
+            </>
+          )}
+          {!showLabel && (
+            <Icon name="arrow" size={showBorder ? "small" : "medium"} />
+          )}
+        </MenuButton>
+      );
+
+      /** Renders all Menu Items to be passed to the Menu List. Flattened into a reduce() so that
+       Groups of items display together. */
+      const getMenuElements = (data: ListItemsData[] = [], selectedItem) =>
```

Comment:
Double check the `selectedItem` argument here. It doesn't seem to be used and is only passed recursively.